aptos-move/block-executor/src/txn_provider/blocking_txns_provider.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::txn_provider::TxnProvider;
+use aptos_mvhashmap::types::TxnIndex;
+use aptos_types::transaction::BlockExecutableTransaction as Transaction;
+use once_cell::sync::OnceCell;
+
+pub struct BlockingTransaction<T: Transaction + std::fmt::Debug> {
+    pub txn: OnceCell<T>,
+}
+
+#[allow(dead_code)]
+impl<T: Transaction + std::fmt::Debug> BlockingTransaction<T> {
+    pub fn new() -> Self {
+        Self {
+            txn: OnceCell::new(),
+        }
+    }
+}
+
+pub struct BlockingTxnsProvider<T: Transaction + std::fmt::Debug> {
+    txns: Vec<BlockingTransaction<T>>,
+}
+
+#[allow(dead_code)]
+impl<T: Transaction + std::fmt::Debug> BlockingTxnsProvider<T> {
+    pub fn new(txns: Vec<BlockingTransaction<T>>) -> Self {
```

Comment:
Can we pass in `num_txns: usize` here instead of the vector and have `new` create the vector internally? Anyway the caller uses `set_txn` and so wouldn't need to access `BlockingTransaction` directly.

Then we can just remove `BlockingTransaction` and internally use a `OnceCell`

Reply:
Done